### PR TITLE
Create stages thunks

### PR DESCRIPTION
**Pull request by Graywing13**

### CODE REVIEW

File: client/src/redux/reducers/stage/thunks.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      return rejectWithValue({ message: error.response.data.error })
+    }
+  },
+)
+
+export const updateStageAsync = createAsyncThunk(
+  actionTypes.UPDATE_STAGE,
+  async ({ id, tripData }, { rejectWithValue }) => {
+    try {
+      const response = await stageService.updateStage(id, tripData)
```
Comment (NLmeng) [inline client/src/redux/reducers/stage/thunks.js]: should probably be stageData?

File: client/src/redux/reducers/stage/service.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import axios from 'axios'
+import { API_URL } from '../../../constants/API_URL'
+
+export const fetchStage = async (id) => {
+  return await axios.get(`${API_URL}/stages/${id}`)
+}
+
+export const fetchStagesByTripId = async (tripId) => {
+  return await axios.get(`${API_URL}/stages`, { tripId: tripId })
+}
```
Comment (NLmeng) [inline client/src/redux/reducers/stage/service.js]: with axios you cant pass tripId like this. need to use their config:
 ``` return await axios.get(`${API_URL}/stages`, { params: { tripId: tripId } })```
 or 
 ``` return await axios.get(`${API_URL}/stages`, { params: { tripId } })```
https://axios-http.com/docs/req_config

File: client/src/redux/reducers/stage/thunks.js (hunk 1 above)
Comment (Graywing13) [reply client/src/redux/reducers/stage/thunks.js]: oo tanku

File: client/src/redux/reducers/stage/service.js (hunk 2 above)
Comment (Graywing13) [reply client/src/redux/reducers/stage/service.js]: ohhh no wonder i was getting a warning. fixed, thanks!

**Review by NLmeng — APPROVED** [review]

**Review by AndyLiang1 — APPROVED** [review]

**Pull request merged**